frontend/app/routes/public/address-validation.tsx

```diff
@@ -128,44 +131,81 @@ export async function action({ context: { configProvider, session }, request }:
     } as const);
   }
 
+  // Get country and provinceState from services
+  const parsedAddress = {
+    address: parsedDataResult.data.address,
+    apartment: parsedDataResult.data.apartment,
+    city: parsedDataResult.data.city,
+    country: serviceProvider.getCountryService().getLocalizedCountryById(parsedDataResult.data.country, locale).name,
+    postalZipCode: parsedDataResult.data.postalZipCode,
+    provinceState: parsedDataResult.data.provinceState && serviceProvider.getProvinceTerritoryStateService().getLocalizedProvinceTerritoryStateById(parsedDataResult.data.provinceState, locale).abbr,
+  };
+
   return {
-    status: 'valid-input',
-    address: parsedDataResult.data,
+    status: 'valid-address',
+    parsedAddress,
   } as const;
 }
 
 export default function AddressValidationRoute() {
   const { t } = useTranslation(handle.i18nNamespaces);
   const { CANADA_COUNTRY_ID, countries, csrfToken, provinceTerritoryStates, USA_COUNTRY_ID } = useLoaderData<typeof loader>();
-  const fetcher = useFetcher<typeof action>();
+  const formElementRef = useRef<HTMLFormElement>(null);
+  const [formKey, setFormKey] = useState(randomString(16));
+  const fetcher = useFetcher<typeof action>({ key: formKey });
```

Comment:
I'm curious about the use of `formKey`. I read through the [`useFetcher`'s `key` documentation](https://remix.run/docs/en/main/hooks/use-fetcher#key) and am still unclear why it's necessary in this specific context. Are there any issues/improvements the use of `formKey` addresses?

Reply:
Away to remount the Form component and clear fetcher's data.
https://github.com/remix-run/remix/discussions/2749#discussioncomment-7372715

Reply:
Seems obvious now - thanks.